### ISSUE TITLE
fit_generator() validation batch size

### ISSUE THREAD

**Issue by MLRadfys** [keras-team/keras#13728]
Hi,

Iam training a U-net with keras using the fit_generator() method. I wrote a custom generator for which I use as a training generator and a validation generator. Now I noticed that I trained with a batch size of 32, but my validation generator was set to a batch size of 1.

I got really nice results with this model, but when I changed the validation batch size to 32, Iam not able to train my model anymore. The validation loss is much higher and after a while the model starts to overfit. 

My question now is: How does the fit_generator computes the MSE metrics and why is my validation loss so low with a batch size of 1, but high with a batch size of 32. Shouldn't the MSE be the same?

Cheers,
Michael

**Comment by karanjakhar** [comment]
There is nothing wrong with the validation generator. MSE is a loss function which takes predicted and expected results as input and returns the loss.

**Comment by MichaelLempart** [comment]
Thanks for your comment.
What I meant is that when you set the batch size to 1 and use the fit_generator() function, keras only returns the mse of the last batch, not the mean over all batches.

**Comment by Saduf2019** [comment]
@MLRadfys 
Moving this issue to closed status as there has been no activity, in case you still face the error please create a new issue.
